### ISSUE TITLE
Why two tails file generated in storage path 

### ISSUE THREAD

**Issue by abhimin** [hyperledger/aries-cloudagent-python#2004]
When I create credential definition with support revocation true then I see two tails files generated in storage path. 
Can you help me to understand that why two files generated and also after revocation another two files generated.

Also let me know how holder connect with tailsserver to get the tails file and how holder use this tails file when presenting to verifier

**Comment by swcurran** [comment]
Here's the background.

Instead of the controller doing all the handling of the RevReg, ACA-Py does it for you.  You say "use revocation" when creating a Cred Def, and the size of the RevReg registry, and ACA-Py does the rest.

Two RevRegs are created initially so that it there is no slowdown when transitioning from a full to an empty RevReg.  Two get created at the start. You issue credentials until the first one is full. ACA-Py automagically switches to using the second one, and creates a new one for when the second one fills up. Repeat.

When your controller revokes a bunch of credentials, ACA-Py will know what RevRegs the credentials are in, and then generate a RevRegEntry transaction for each .

Bottom line -- your controller just says "I'm using revocation" and revokes credentials when needed. ACA-Py takes care of all the messy issues in doing that.

**Comment by swcurran** [comment]
Tails files -- the RevRegDef transaction has the location of a URL for the tails file, and the holder uses that to get the file -- usually via an HTTPS request (assuming that is what is in the URL).

How the tails file is used by the holder is described here: https://github.com/hyperledger/indy-hipe/tree/main/text/0011-cred-revocation
